tests/test_tokenization_common.py

```diff
@@ -2232,6 +2252,10 @@ def test_alignement_methods(self):
                     self.assertIn(None, pair_batch_sequence_ids)
 
     def test_tokenization_python_rust_equals(self):
+        if self.tokenizer_class is None:
+            # as we don't have a slow version, we can't compare the outputs between slow and fast versions
+            return
+
         for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
```

Comment:
This doesn't originate from your PR, but it's a bit weird that `self.tokenizers_list = []` when there is no Rust tokenizer.

I would expect a check to ensure we're testing fast tokenizers, which would make sense used alongside your added code to check that the `self.tokenizer_class` is not `None`; but here we're having two different approaches which are unnecessarily hard to understand: the `self.tokenizers_list` is defined according to if we test the fast tokenizer, while the test is skipped according to if we test the slow tokenizer.

I would expect either `self.tokenizers_list` to also be `[]` when `self.tokenizer_class` isn't defined, or for the test to be skipped on two similar conditions. Does that make sense?

Reply:
I have just realized that actually this `self.tokenizers_list` is equal to `[]` only if the rust tokenizer isn't tested, and the python tokenizer is. If both are tested, then this contains both, and if only the rust is tested, then only the rust tokenizer is in it.

Would it make sense to have the `self.tokenizers_list` always contain the tokenizers available, irrespective to the availability of one or the other?

Reply:
I agree with you that it is complicated to reconstruct what is being tested here.

I had noticed that the `self.tokenizers_list` attribute is used in the tests in (almost) 3 different ways (listed in the table [here](https://docs.google.com/spreadsheets/d/12hTT2gEf8ZM0OjA4CvDP_DM0wXvPcQugg6-oCKjcK3w/edit?usp=sharing)):
1. By instantiating only a rust tokenizer(s);
2. By instantiating rust tokenizer(s) and slow tokenizer(s) and comparing in asserts the `outputs` of these 2 types of tokenizers;
3. By instantiating a rust tokenizer(s) and a slow tokenizer(s) and by never directly comparing in asserts the `outputs` of these 2 types of tokenizers;
4. (this happens, unless I'm mistaken, just in the `test_save_pretrained` method) By instantiating a rust tokenizer(s) and a slow tokenizer(s) and making the 2 types of asserts.

Reply:
Before making a new proposal on this topic, what I'm not sure I've figured out yet is in what case we would have `self.tokenizer_class` different from `None` and `test_rust_tokenizer` set to `False`.

Reply:
Thank you for putting together that table, that's helpful. I don't think I see such a case in the current tests!